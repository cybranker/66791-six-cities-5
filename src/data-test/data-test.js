export const comment = {
  comment: `What an amazing view! The house is stunning and in an amazing location. The large glass wall had an amazing view of the river!`,
  date: `2020-10-25T13:38:44.753Z`,
  id: 1,
  rating: 3,
  user: {
    [`avatar_url`]: ``,
    id: 13,
    [`is_pro`]: false,
    name: `Zak`
  }
};
