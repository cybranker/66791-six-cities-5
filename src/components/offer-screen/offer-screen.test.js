import React from "react";
import renderer from "react-test-renderer";
import {Provider} from "react-redux";
import {MemoryRouter} from "react-router";
import OfferScreen from "./offer-screen";
import createStore from "../../mocks/mock-store";
import {offers, comments, user, matchID} from "../../data-test/data-test";

const noop = () => {};

test(`Should OfferScreen render correctly`, () => {
  const tree = renderer
    .create(
        <Provider store={createStore()}>
          <MemoryRouter>
            <OfferScreen
              loadOffer={noop}
              loadComments={noop}
              loadOffersNearby={noop}
              offer={offers[1]}
              reviews={comments}
              nearby={offers}
              changeOfferActiveAction={noop}
              offerActive={{}}
              authorizationStatus={`AUTH`}
              user={user}
              redirectLoginClick={noop}
              onClickAddFavorite={noop}
              match={matchID}
            />
          </MemoryRouter>
        </Provider>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
