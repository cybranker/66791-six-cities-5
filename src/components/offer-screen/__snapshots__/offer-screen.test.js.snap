// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should OfferScreen render correctly 1`] = `""`;
